### PR TITLE
Add unit tests for `add` command

### PR DESCRIPTION
**Pull request by arielireni**
This PR includes:
- Unit tests for the `add` command
- Changes after `go mod tidy` & `go mod vendor`, and `go install` for ginkgo and gomega.
- Fix a typo in the `add workload` command implementation.

Signed-off-by: arielireni <aireni@redhat.com>

### CODE REVIEW

File: internal/cmd/add/add_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+			Expect(err).NotTo(BeNil())
+
+			// then
+			Expect(actualErr.String()).To(Equal("Error: required flag(s) \"name\" not set\n"))
+		})
+
+
+		It("should fail to add a new workload without a device", func() {
+			// given
+			rootCmd.SetArgs([]string{"add", "workload"})
```
Comment (pkliczewski) [inline internal/cmd/add/add_test.go:143]: We are missing `workload1` here as well. Is this expected not to have it in error message?

File: internal/cmd/add/add_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+			Expect(err).To(BeNil())
+
+			// then
+			Expect(actualOut.String()).To(Equal("device 'device1' was added\n"))
+			Expect(actualErr.String()).To(BeEmpty())
+		})
+
+		It("should add a new workload", func() {
+			// given
+			rootCmd.SetArgs([]string{"add", "workload", "-d", "device1", "-n", "workload1"})
```
[the remark this answers is not in the document]
Comment (arielireni) [reply internal/cmd/add/add_test.go:82]: yes, I used ginkgo's [ordered containers](https://onsi.github.io/ginkgo/#ordered-containers) option in this context so the test which should add the device will run before this test (and if the first test will fail we'll skip this test)

File: internal/cmd/add/add_test.go (hunk 1 above)
Comment (arielireni) [reply internal/cmd/add/add_test.go:143]: If the workload's name were not provided, we would use the image + random suffix as its name (we allow to add workloads without providing their names [here](https://github.com/project-flotta/flotta-dev-cli/blob/main/internal/cmd/add/workload.go#L61))

File: internal/cmd/add/add_test.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+			actualOut.Reset()
+			actualErr.Reset()
+		})
+
+		AfterAll(func() {
+			client, err := resources.NewClient()
+			Expect(err).To(BeNil())
+
+			// remove test device
+			device, err := resources.NewEdgeDevice(client, "device1")
```
[the remark this answers is not in the document]
Comment (arielireni) [reply internal/cmd/add/add_test.go]: Replaced with removing the workload while we remove the device, WDYT?

File: internal/cmd/add/add_test.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+			Expect(err).To(BeNil())
+
+			// remove test workload
+			workload, err := resources.NewEdgeWorkload(client)
+			Expect(err).To(BeNil())
+			err = workload.Remove("workload1")
+			Expect(err).To(BeNil())
+
+			// remove test device-set
+			deviceset, err := resources.NewEdgeDeviceSet(client, "set1")
```
[the remark this answers is not in the document]
Comment (arielireni) [reply internal/cmd/add/add_test.go]: Done

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**